### PR TITLE
scaffold contract

### PR DESCRIPTION
**Pull request by alloystory**

### CODE REVIEW

File: packages/hardhat/contracts/WagmiContract.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+contract WagmiContract {
+  struct Listing {
+    address tokenAddr;
+    uint256 tokenId;
+    address ownerAddr;
+    uint256 listPrice;
+    uint256 promoterReward;
+    uint256 buyerReward;
+  }
```
Comment (andyrobert3) [inline packages/hardhat/contracts/WagmiContract.sol:17]: just to be on the same page
- `tokenAddr` -> NFT address
- `tokenId` internal id we use for listing the NFT ("/:tokenId")
- `ownerAddr` -> token owner
- `listPrice` -> price of token to be listed
- `promoterReward` -> how much will the promoter get by selling, given by the client
- `buyerReward`

File: packages/hardhat/contracts/WagmiContract.sol [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  Listing[] listings;
+  mapping(address => bool) promoters;
```
Comment (andyrobert3) [inline packages/hardhat/contracts/WagmiContract.sol:20]: is this just to check if the `address` is a promoter

**Review by andyrobert3 — APPROVED** [review]

File: packages/hardhat/contracts/WagmiContract.sol (hunk 2 above)
Comment (alloystory) [reply packages/hardhat/contracts/WagmiContract.sol:20]: yeap

File: packages/hardhat/contracts/WagmiContract.sol (hunk 1 above)
Comment (alloystory) [reply packages/hardhat/contracts/WagmiContract.sol:17]: `tokenAddr` -> NFT contract addr
`tokenId` -> id of the token on that NFT contract's address

reward -> yes but in percentage instaed of raw amount

**Pull request merged**